### PR TITLE
Gets rid of an annoying sporadic CI failure on closets by actually properly checking if they're qdeleted or not, no matter their typepath

### LINKED ISSUE #69732: Spurious test failure: /obj/structure/closet/emcloset/anchored was unable to be GC'd
## Reproduction:

```
 ## REF SEARCH Beginning search for references to a /obj/structure/closet/emcloset/anchored.
## REF SEARCH Finished searching globals
## REF SEARCH Finished searching native globals
## REF SEARCH Found /obj/structure/closet/emcloset/anchored [0x2003844] in /obj/item/tank/internals/emergency_oxygen's [0x2003845] loc var. World -> /obj/item/tank/internals/emergency_oxygen
## REF SEARCH Found /obj/structure/closet/emcloset/anchored [0x2003844] in list World -> /obj/item/tank/internals/emergency_oxygen [0x2003845] -> locs (list).
## REF SEARCH Found /obj/structure/closet/emcloset/anchored [0x2003844] in /obj/item/clothing/mask/breath's [0x200384a] loc var. World -> /obj/item/clothing/mask/breath
## REF SEARCH Found /obj/structure/closet/emcloset/anchored [0x2003844] in list World -> /obj/item/clothing/mask/breath [0x200384a] -> locs (list).
## REF SEARCH Finished searching atoms
## REF SEARCH Finished searching datums
## REF SEARCH Finished searching clients
## REF SEARCH Completed search for references to a /obj/structure/closet/emcloset/anchored.
## TESTING: GC: -- [0x2003844] | /obj/structure/closet/emcloset/anchored was unable to be GC'd --
Error: /obj/structure/closet/emcloset/anchored hard deleted 1 times out of a total del count of 4
FAIL: /datum/unit_test/create_and_destroy 536s
	REASON #1: /obj/structure/closet/emcloset/anchored hard deleted 1 times out of a total del count of 4 at code/modules/unit_tests/create_and_destroy.dm:167
```

https://github.com/tgstation/tgstation/runs/8196398955?check_suite_focus=true

### PR DESCRIPTION
**Pull request by GoldenAlpharex**
## About The Pull Request
The bug actually came from the `closets.dm` unit test, where we were using an `istype()` on a typepath, rather than an `ispath()`. Either way, that check was unnecessary, so I removed it entirely, to make it more future-proof anyway.

Fixes #69732.

## Why It's Good For The Game
I think maintainer sanity is good for the game.

## Changelog
🆑 GoldenAlpharex
fix: Emergency closets will no longer make maintainers cry due to causing sporadic unit test failures, because the unit test was doing the wrong kind of check on a typepath.
/🆑

### CODE REVIEW

**Comment by Mothblocks** [comment]
Fixing CI failures by removing features is not allowed

**Comment by Mothblocks** [comment]
Also @LemonInTheDark told me the test doesn't even fail when the closets delete themselves 100% of the time so

**Comment by GoldenAlpharex** [comment]
> Fixing CI failures by removing features is (generally) not allowed

I'd like to discuss this some more with you, where's the best place to reach out to you considering you're not currently on the /tg/ Discord?

**Comment by tralezab** [comment]
functionality preserved in a neat way

**Comment by Mothblocks** [comment]
I'm on coderbus

File: code/game/objects/structures/crates_lockers/closets/utility_closets.dm [hunk 1]
```diff
@@ -20,14 +20,14 @@
...
-
-	if (prob(1))
-		return INITIALIZE_HINT_QDEL
+#define PROB_SPAWN_EMPTY 1
 
 /obj/structure/closet/emcloset/PopulateContents()
-	..()
+	. = ..()
+
+	// Let's just make it spawn empty, rather than making it queue itself for deletion on Initialize().
```
Comment (Mothblocks) [inline code/game/objects/structures/crates_lockers/closets/utility_closets.dm:28]:
```suggestion

```

No historical comments

**Comment by Mothblocks** [comment]
I have been informed (by you) this is just basically the same thing since it already had a chance to be empty and now I'm unhappy again

Work with Lemon on figuring out what the actual issue is

**Comment by GoldenAlpharex** [comment]
Reverting that second commit since it didn't serve any actual purpose anymore. I'll see what else we do.

**Comment by LemonInTheDark** [comment]
This should be supported, even if it's silly
if it's broken, we should fix that, rather then leaving a mantrap
I want to close this

**Comment by LemonInTheDark** [comment]
> Also @LemonInTheDark told me the test doesn't even fail when the closets delete themselves 100% of the time so

It did on my local. It's fukin weird it hurts

**Comment by GoldenAlpharex** [comment]
This is probably the weirdest commit history I've ever had on a PR.

**Comment by msgerbs** [comment]
I'm like 99% sure this just how it was a week ago before the 8th attempt to fix this.

**Comment by GoldenAlpharex** [comment]
> I'm like 99% sure this just how it was a week ago before the 8th attempt to fix this.

Not quite, actually. Back then, the check wasn't there at all. Then, the check was added, but it was done wrong, and over-complicated. Now, it's simplified, and should actually do what it was meant to do, and catch it when a locker was qdel'd.

**Pull request merged**